File: Data Structures & Algorithms/car-fleet/submission-1.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        paired = [(p, s) for p, s in zip(position, speed)]
        paired.sort(reverse = True)

        fleets = 1
        prevCar = (target - paired[0][0]) / paired[0][1]
        for i in range(1, len(paired)):
            currCar = (target - paired[i][0]) / paired[i][1]
            if currCar > prevCar:
                fleets += 1
                prevCar = currCar
        return fleets

File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size() - 1;

        while (l < r){
            int total = numbers[l] + numbers[r];

            if (total > target){
                r--;
            }
            else if (total < target){
                l++;
            }
            else {
                return {l + 1, r + 1};
            }
        }
        return {};
    }
};
